exec/java-exec/src/test/java/org/apache/drill/exec/physical/impl/TestOrderedMuxExchange.java

```diff
@@ -0,0 +1,116 @@
+    long previousBigInt = Long.MIN_VALUE;
+
+    for (QueryDataBatch b : results) {
+      RecordBatchLoader loader = new RecordBatchLoader(allocator);
+      if (b.getHeader().getRowCount() > 0) {
+        loader.load(b.getHeader().getDef(),b.getData());
+        @SuppressWarnings({ "deprecation", "resource" })
+        IntVector c1 = (IntVector) loader.getValueAccessorById(IntVector.class,
+                   loader.getValueVectorId(new SchemaPath("id_i", ExpressionPosition.UNKNOWN)).getFieldIds()).getValueVector();
+        IntVector.Accessor a1 = c1.getAccessor();
+
+        for (int i = 0; i < c1.getAccessor().getValueCount(); i++) {
+          assertTrue(String.format("%d > %d", previousBigInt, a1.get(i)), previousBigInt <= a1.get(i));
+          previousBigInt = a1.get(i);
+        }
+      }
+      loader.clear();
+      b.release();
+    }
+  }
+
+  /**
+   * Test case to verify the OrderedMuxExchange created for order by clause.
+   * It checks by forcing the plan to create OrderedMuxExchange and also verifies the
+   * output column is ordered.
+   *
+   * @throws Exception if anything goes wrong
+   */
+
+  @Test
+  public void testOrderedMuxForOrderBy() throws Exception {
+    ClusterFixtureBuilder builder = ClusterFixture.builder(dirTestWatcher)
+            .maxParallelization(1)
+            .configProperty(ExecConstants.SYS_STORE_PROVIDER_LOCAL_ENABLE_WRITE, true)
+            ;
+
+    try (ClusterFixture cluster = builder.build();
+         ClientFixture client = cluster.clientFixture()) {
+      client.alterSession(ExecConstants.SLICE_TARGET, 10);
+      String sql = "SELECT id_i, name_s10 FROM `mock`.`employees_10K` ORDER BY id_i limit 10";
+
+      String explainText = client.queryBuilder().sql(sql).explainText();
+      assertTrue(explainText.contains(ORDERED_MUX_EXCHANGE));
+      validateResults(client.allocator(), client.queryBuilder().sql(sql).results());
+    }
+  }
+
+  /**
+   * Test case to verify the OrderedMuxExchange created for window functions.
+   * It checks by forcing the plan to create OrderedMuxExchange and also verifies the
+   * output column is ordered.
+   *
+   * @throws Exception if anything goes wrong
+   */
+
+  @Test
+  public void testOrderedMuxForWindowAgg() throws Exception {
+    ClusterFixtureBuilder builder = ClusterFixture.builder(dirTestWatcher)
+            .maxParallelization(1)
+            .configProperty(ExecConstants.SYS_STORE_PROVIDER_LOCAL_ENABLE_WRITE, true)
+            ;
+
+    try (ClusterFixture cluster = builder.build();
+         ClientFixture client = cluster.clientFixture()) {
+      client.alterSession(ExecConstants.SLICE_TARGET, 10);
+      String sql = "SELECT id_i, max(name_s10) over(order by id_i) FROM `mock`.`employees_10K` limit 10";
+
+      String explainText = client.queryBuilder().sql(sql).explainText();
+      assertTrue(explainText.contains(ORDERED_MUX_EXCHANGE));
+      validateResults(client.allocator(), client.queryBuilder().sql(sql).results());
```

Comment:
I think a simpler way to validate results would be to compare with a plan that does not have this mux exchange; that will allow arbitrary data types to be handled since currently you have a hardcoded IntVector comparison which limits the types of queries one can test.